keras_nlp/models/xlm_roberta/xlm_roberta_preprocessing.py

```diff
@@ -169,3 +159,87 @@ def from_preset(
         **kwargs,
     ):
         raise NotImplementedError
+
+
+class XLMRobertaTokenizer(SentencePieceTokenizer):
+    """XLM-RoBERTa tokenizer layer.
+
+    This layer provides as implementation of XLM-RoBERTa tokenization based on
+    `keras_nlp.tokenizers.SentencePieceTokenizer`. This layer is necessary
+    because the original fairseq implementation modifies the indices of the
+    SentencePiece tokenizer output. To preserve compatibility, we make the same
+    changes, i.e., `"<s>"`, `"<pad>"`, `"</s>"` and `"<unk>"` are mapped to
+    0, 1, 2, 3, respectively, and non-special tokens' indices are shifted right
+    by one.
+
+    Note: This is an unexported tokenizer, meant to be used in
```

Comment:
I don't think we actually need to document this. We choose what to export by what we put in out `__init__.py` files and keras.io.

Eventually we may actually go to a keras style generated API. Where we have a line `keras_export` the controls our API symbols.